fabric-content-registries-v0/src/main/java/net/fabricmc/fabric/api/registry/LandPathNodeTypesRegistry.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.registry;
+
+import java.util.IdentityHashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.ai.pathing.PathNodeType;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.BlockView;
+
+/**
+ * A registry to associate specific node types to blocks.
+ * Specifying a node type for a block will change the way an entity recognizes the block when trying to pathfind.
+ * For example, you can specify that a block is dangerous and should be avoided by entities.
+ * This works only for entities that move on air and land.
+ */
+public final class LandPathNodeTypesRegistry {
+	private static final Logger LOGGER = LoggerFactory.getLogger(LandPathNodeTypesRegistry.class);
+	private static final Map<Block, PathNodeTypeProvider> NODE_TYPES = new IdentityHashMap<>();
+
+	private LandPathNodeTypesRegistry() {
+	}
+
+	/**
+	 * Registers a {@link PathNodeType} for the specified block.
+	 * This will override the default block behavior.
+	 * For example, you can make a safe block as dangerous and vice-versa.
+	 * Duplicated registrations for the same block will replace the previous registration.
+	 *
+	 * @param block              Block to register.
+	 * @param nodeType           {@link PathNodeType} to associate with the block.
+	 *                           (Pass {@code null} to not specify a node type and use the default behavior)
+	 * @param nodeTypeIfNeighbor {@link PathNodeType} to associate to the block, if is a neighbor block in the path.
```

Comment:
Sentence grammar and consistency of "associate with"

```suggestion
	 * @param nodeTypeIfNeighbor {@link PathNodeType} to associate with the block, if it is a neighbor block in the path.
```